### ISSUE TITLE
Output discrepancy between keras.predict and tf_saved_model

### ISSUE THREAD

**Issue by edge7** [keras-team/keras#20985]
Hi,
I am exporting a model like this:

```
       import keras

        export_archive = keras.export.ExportArchive()
        export_archive.track(model_tra)
        export_archive.add_endpoint(
            name="serve",
            fn=lambda x: model_tra.call(x, training=False),
            input_signature=[
                keras.InputSpec(shape=(None, 3, 320, 320, 1), dtype="float64")
            ],
        )
        export_archive.write_out(
            "/home/edge7/Desktop/projects/ing_edurso/wharfreenet-inference/models/la_ao"
        )
```

Then I load it like:
`tf.saved_model.load(os.path.join(MODEL_DIRECTORY, "la_ao"))`

and I use it like:

`model.serve(np.expand_dims(np.array(frame), axis=-1))`

I get slightly different results than just using:
model.predict or model.predict_on_batch

am I missing something silly in the conversion?

**Comment by SamanehSaadat** [comment]
Another question: Do you get different results calling `model.predict` or `model.predict_on_batch` right after training the model, i.e. without going through the export process?

**Comment by edge7** [comment]
Hi @SamanehSaadat 
[Please find here ](https://colab.research.google.com/drive/1ROYIeovLrc1VkS4Ip-wtr723lB1i7GaR?usp=sharing) a colab, that is self contained.
It downloads the models plus the model definition on the fly.
Then it tests with Keras (Jax), tf_model, and onnx.
